LG-2937: bin/activate will download application.yml from S3 and merge in

solipet opened the pull request:
In looking to mirror the way the IDP works, I finally took a look at the application.yml's that are up on `dev` and `int` and I'd like to start working on cleaning them up.

I think about 40% of the settings could just live in `application.yml.example` since they aren't secret or different across the environments. I included these settings in `application.yml.example` directly.

The configs stored up on S3 currently have all settings in alphabetical order. I'd like to remove the common settings I've included in the example config, then list the remainder in alphabetical order, but broken out into a few groups: feature flags, secrets, overrides, etc.

Finally, in the example config, we have settings for `saml_sp_certificate`, `saml_sp_private_key`, and `saml_sp_private_key_password`. These are listed as required in `config/initializers/figaro.rb`, but are not set in either the `dev` or `int` application.yml's. The certificate is not referenced anywhere in the codebase. The key and password are used in the omniauth config, but that's an OIDC config, so...

zachmargolis commented on lib/deploy/activate.rb:
```diff
@@ -0,0 +1,64 @@
+    end
+
+    def set_proper_file_permissions_for_application_yml
+      FileUtils.chmod(0o640, [env_yaml_path, result_yaml_path])
```
oooh fancy octal notation, I didn't realize this was in the IDP too

zachmargolis commented on spec/spec_helper.rb:
```diff
@@ -12,6 +12,7 @@
 # http://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
 RSpec.configure do |config|
   config.color = true
+  config.formatter = :documentation
```
😍

zachmargolis commented on config/application.yml.example:
```diff
@@ -1,3 +1,44 @@
+enable_load_testing_mode: 'true'
+enable_mobile_capture: 'true'
+hmac_fingerprinter_key_queue: '["old-key-one", "old-key-two"]'
+in_person_proofing_enabled: 'false'
```
this is not a dashboard feature flag as far as I can tell....should we remove it?

zachmargolis commented on config/application.yml.example:
```diff
@@ -1,3 +1,44 @@
+allow_doc_auth_test_credentials: 'true'
+attribute_encryption_key_queue: '["old-key-one", "old-key-two"]'
+aws_region: us-west-2
+cac_proofing_enabled: 'true'
```
ditto .... this is an IDP feature flag, I think we should remove it here

jmhooper replied:
Yeah, I think there are quite a few IdP feature flags in here 🤔

solipet replied:
Ugh.. I think I mixed up the idp application.yml!! 🤦

solipet commented:
@zachmargolis @jmhooper I've cleaned this up and uploaded the current `application.yml` in use on dev/int to s3 in preparation of this becoming the new way.

solipet commented:
> LGTM, looks like a spec failure, but once that get resolved go ahead and merge:
> 
> ```
>      # ./spec/features/logo_upload_spec.rb:26:in `block (3 levels) in <top (required)>'
> ```
> 
> Maybe related to `fakefs` messing with filesystem access?

I really don't understand what changed between the passing test and now that could cause a consistent failure.  `git diff c877402 78f5356`. I cleaned up the example yml file. I moved `smtp.rb` from `config/` to `config/initializers/` and told `production.rb` where to find it. I removed the requirement to have `saml_sp_certificate` in `figaro.rb` since we no longer use it. And finally, I updated the `activate_spec` to look for settings that are actually in the example yml now. (╯ಠ益ಠ)╯彡 ┻━┻

solipet commented:
> > LGTM, looks like a spec failure, but once that get resolved go ahead and merge:
> > ```
> >      # ./spec/features/logo_upload_spec.rb:26:in `block (3 levels) in <top (required)>'
> > ```
> > 
> > 
> > Maybe related to `fakefs` messing with filesystem access?
> 
> I really don't understand what changed between the passing test and now that could cause a consistent failure. `git diff c877402 78f5356`. I cleaned up the example yml file. I moved `smtp.rb` from `config/` to `config/initializers/` and told `production.rb` where to find it. I removed the requirement to have `saml_sp_certificate` in `figaro.rb` since we no longer use it. And finally, I updated the `activate_spec` to look for settings that are actually in the example yml now. (╯ಠ益ಠ)╯彡 ┻━┻

Bah!, nevermind -- found it 😝
